ostd/src/trap/handler.rs

```diff
@@ -26,6 +26,13 @@ pub(crate) fn call_irq_callback_functions(trap_frame: &TrapFrame, irq_number: us
 
     crate::arch::irq::enable_local();
     crate::trap::softirq::process_pending();
+
+    // TODO: add preemption point after handling irqs once we have a reasonable policy to
```

Comment:
No I don't like this `TODO` either. We should only call `schedule` on certain timer callbacks, rather than all IRQ events. We do preemption checks regularly, not every time the user presses the keyboard.

Reply:
I might be sleepy or drunk at the time of writing this line and I don't know what I was trying to express. But this is still stuff about kernel preemption. I'll remove this line since we are now far from ready for that.